descqa/DeltaSigmaTest.py

```diff
@@ -0,0 +1,118 @@
+
+        mask_lowz = np.abs(cperp) < 0.2 # color boundaries
+        mask_lowz &= res['mag_true_r_sdss'] < (13.5 + cpar/0.3) # sliding magnitude cut
+        mask_lowz &= (res['mag_true_r_sdss'] > 16) &(res['mag_true_r_sdss'] < 19.6)
+
+        # Counting the number density of LOWZ galaxies
+        nlens = len(np.where(mask_lowz)[0]) / catalog_instance.sky_area
+        with open(os.path.join(output_dir, 'galaxy_density.dat'), 'a') as f:
+            f.write('{} \n'.format(nlens))
+
+        #  Additional redshift cuts used in Singh et al. (2015)
+        mask_lowz &= (res['redshift_true'] > 0.16) & (res['redshift_true'] < 0.36)
+
+        # Computing mask for source sample, this only serves to keep the number
+        # of neighbours manageable
+        mask_source = (res['mag_true_i_sdss'] < 26.) & (res['redshift_true'] > 0.4)
+
+        # Create astropy coordinate objects
+        coords = SkyCoord(ra=res['ra']*u.degree, dec=res['dec']*u.degree)
+        coords_l = coords[mask_lowz]
+        coords_s = coords[mask_source]
+
+        # Search for neighbours
+        idx1, idx2, sep2d, _ = search_around_sky(coords_l, coords_s, 2.*u.deg)
+
+        # Computing sigma crit for each pair
+        zl = res['redshift_true'][mask_lowz][idx1]
+        zs = res['redshift_true'][mask_source][idx2]
+
+        # Warning: this assumes a flat universe
+        # See http://docs.astropy.org/en/v0.3/_modules/astropy/cosmology/core.html#FLRW.angular_diameter_distance_z1z2
+        dm1 = self.comoving_transverse_distance(zl)
+        dm2 = self.comoving_transverse_distance(zs)
+        angular_diameter_distance_z1z2 = u.Quantity((dm2 - dm1)/(1. + zs), u.Mpc)
+
+        sigcrit = cst.c**2 / (4.*np.pi*cst.G) * self.angular_diameter_distance(zs) / \
+                ((1. + zl)**2. * angular_diameter_distance_z1z2 * self.angular_diameter_distance(zl))
+        # Apply unit conversion to obtain sigma crit in h Msol /pc^2
+        cms = u.Msun / u.pc**2
+        sigcrit = sigcrit*(u.kg/(u.Mpc* u.m)).to(cms) / 0.7
+
+        # Computing the projected separation for each pairs, in Mpc/h
+        r = sep2d.rad*self.angular_diameter_distance(zl)*(1. + zl) * 0.7
+
+        # Computing the tangential shear
+        thetac = np.arctan2(
+            (coords_s[idx2].dec.rad - coords_l[idx1].dec.rad) / np.cos((coords_s[idx2].dec.rad + coords_l[idx1].dec.rad) / 2.0),
+            coords_s[idx2].ra.rad - coords_l[idx1].ra.rad
+        )
+        gammat = -(res['shear_1'][mask_source][idx2] * np.cos(2*thetac) - res['shear_2'][mask_source][idx2] * np.sin(2*thetac))
+
+        # Binning the tangential shear
+        bins = np.logspace(np.log10(0.05), 1, 17, endpoint=True)
+        counts = np.histogram(r, bins=bins)[0]
+        gt, b = np.histogram(r, bins=bins, weights=gammat*sigcrit)
+        rp = 0.5*(b[1:]+b[:-1])
+
+        counts[counts < 1] = 1
+        gt = gt / counts
```

Comment:
I recommend issuing a warning if `counts` is below a certain value (like 10 or 100).  It indicates that the test is being misused.

Reply:
not sure about a warning, but we can fail the test if the number density gets outside of this  range.

Reply:
I've added code that will fail the test if the minimum number of bakground galaxies in a radial bin falls below a given number

Reply:
thanks @yymao